File: frontend/src/app/core/services/response/room.response.ts
import { RoomStatus } from '../../models/room-status.enum';
import { UserResponse } from './user.response';

export interface RoomResponse {
  id: number;
  name: string;
  description?: string;
  status: RoomStatus;
  createdBy: UserResponse;
  createdAt: string;
}

File: backend/src/main/java/com/finki/backend/web/response/RoomResponse.java
package com.finki.backend.web.response;

import com.finki.backend.core.domain.enums.RoomStatus;
import lombok.Builder;

import java.time.Instant;

@Builder
public record RoomResponse(
        Long id,
        String name,
        String description,
        RoomStatus status,
        UserResponse createdBy,
        Instant createdAt
) {
}
